app/models/attachment_data.rb

```diff
@@ -103,6 +103,15 @@ def replace_with!(replacement)
     end
   end
 
+  def uploaded_to_asset_manager!
+    update!(uploaded_to_asset_manager_at: Time.zone.now)
+    ServiceListeners::AttachmentUpdater.update_attachment_data! self
+  end
+
+  def uploaded_to_asset_manager?
+    uploaded_to_asset_manager_at != nil
```

Comment:
Surprised the linter didn't make you do `uploaded_to_asset_manager_at.present?`.